test/core/test_lwt_mvar.ml

```diff
@@ -0,0 +1,73 @@
+(* OCaml promise library
+ * http://www.ocsigen.org/lwt
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, with linking exceptions;
+ * either version 2.1 of the License, or (at your option) any later
+ * version. See COPYING file for details.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+ * 02111-1307, USA.
+ *)
+
+open Test
+open Lwt
+
+let suite =
+  suite "lwt_mvar" [
+    test "basic take"
+      (fun () ->
+         let x = Lwt_mvar.create 0 in
+         let y = Lwt_mvar.take x in
+         return (y = Lwt.return 0)
+      );
+
+    test "blocking put"
+      (fun () ->
+         let x = Lwt_mvar.create 0 in
+         let y = Lwt_mvar.put x 1 >>= (fun _ -> return 1) in
+         return (y <> Lwt.return 1)
+      );
+
+    test "put-take"
+      (fun () ->
+         let x = Lwt_mvar.create_empty () in
+         let z = Lwt_mvar.put x 0 >>= (fun _ -> Lwt_mvar.take x) in
+         return (z = Lwt.return 0)
+      );
+
+    test "take-put"
+      (fun () ->
+         let x = Lwt_mvar.create 0 in
+         let _ = Lwt_mvar.take x in
+         let z = Lwt_mvar.put x 1 >>= (fun _ -> return 2) in
+         return (z = Lwt.return 2)
```

Comment:
The above optional suggestion about `Lwt.state` would save a bind here as well, and the need for `2` :) Another way is to compare with `Lwt.return ()`, but... `Lwt.state` :)